ansible_image_validation/validation-playbooks/per-vm-validation.yaml

```diff
@@ -134,13 +134,138 @@
       line: "Rhui Client package is not installed in the image"
       create: yes
       state: present
-    when: rhui_package_count == "0"
+    when: rhui_package_count == "0" and offer_type !='byol'
 
   - name: Set the variable if rhui client package is found
     set_fact:
       is_rhui_package_present : true
     when: rhui_package_count != "0"
 
+##Repo Validation Task
+
+- name: Check if offertype is equal to byol and client package is present
+  lineinfile:
+    path: "{{err_folder}}/err_msgs.log"
+    line: "\n  Client Package is present in byol Offertype Image"
+    create: yes
+    state: present 
+  when: offer_type =='byol' and rhui_package_count != "0"
+
+- name: Check the value of client package
+  shell: rpm -qa | grep rhui
+  register: output_client_package
+  when: offer_type!='byol'
+
+- debug:
+    var: output_client_package
+
+- name: Check the architecture of the RHEL Image.
+  shell: echo {{ architecture }}
+  register: output_architecture
+
+- debug:
+    var: output_architecture
+
+- name: Check if the client package is present according to the offertype when offertype is not equal to byol and architecture is equal to x86-64.
+  ansible.builtin.shell:
+    cmd: |
+      #!/bin/bash
```

Comment:
explain the logic in description and in comments in the playbook

Reply:
Done